File: src/main/java/plugin/command/player/SetLevelCommandPlugin.java
package plugin.command.player;

import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.content.plugin.type.CommandPlugin;
import org.redrune.utility.constants.SkillConstants;
import plugin.command.CommandManifest;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 9/1/2017
 */
@CommandManifest(description = "Sets your levels", types = { Integer.class, Integer.class })
public class SetLevelCommandPlugin extends CommandPlugin {
	
	@Override
	public void handle(Player player, String[] args, boolean console, boolean clientCommand) {
		int skill = intParam(args, 1);
		int level = intParam(args, 2);
		if (level < 0 || level > 99) {
			player.getPackets().sendMessage("Please choose a valid level.");
			return;
		}
		player.getSkills().set(skill, level);
		player.getSkills().setXp(skill, SkillConstants.getXPForLevel(level));
		player.getAppearance().generateAppearanceData();
	}
	
	@Override
	public String[] identifiers() {
		return arguments("setlevel");
	}
}

File: src/main/kotlin/plugin/command/player/SetLevelCommandPlugin.kt
package plugin.command.player

import plugin.command.CommandManifest
import org.redrune.game.content.plugin.type.CommandPlugin
import org.redrune.game.global.World
import org.redrune.utility.constants.SkillConstants
import plugin.command.player.YellCommandPlugin
import org.redrune.utility.functions.Misc
import org.redrune.game.entity.actor.player.data.PlayerRight
import org.redrune.utility.constants.InterfaceConstants
import org.redrune.cache.loaders.ItemDefinitions
import org.redrune.game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 9/1/2017
 */
@CommandManifest(description = "Sets your levels", types = [Int::class, Int::class])
class SetLevelCommandPlugin : CommandPlugin() {
    override fun handle(player: Player, args: Array<String>, console: Boolean, clientCommand: Boolean) {
        val skill = intParam(args, 1)
        val level = intParam(args, 2)
        if (level < 0 || level > 99) {
            player.packets.sendMessage("Please choose a valid level.")
            return
        }
        player.skills[skill] = level
        player.skills.setXp(skill, SkillConstants.getXPForLevel(level).toDouble())
        player.appearance.generateAppearanceData()
    }

    override fun identifiers(): Array<String> {
        return arguments("setlevel")
    }
}